python_modules/dagster/dagster/_core/definitions/multi_dimensional_partitions.py

```diff
@@ -477,7 +477,7 @@ def time_window_partitions_def(self) -> TimeWindowPartitionsDefinition:
             check.inst(self.primary_dimension.partitions_def, TimeWindowPartitionsDefinition),
         )
 
-    def time_window_for_partition_key(self, partition_key: str) -> TimeWindow:
+    def time_window_for_partition_key(self, partition_key: str) -> PersistedTimeWindow:
```

Comment:
this is an example of a method that i'm not sure is de-facto 'public' in that users might be calling it and changes like this might break them, even if it is not technically so

Reply:
It seemed that it was essentially public based on our own usage of it in integrations so I kept it returning the public version